components/post_view/post_list/post_list_virtualized.jsx

```diff
@@ -443,8 +471,118 @@ class PostList extends React.PureComponent {
 
     scrollToBottom = () => {
         this.listRef.current.scrollToItem(0, 'end');
+        this.setState({renderUnreadChannelToast: false});
+    }
+
+    scrollToUnread = () => {
+        this.listRef.current.scrollToItem(this.getNewMessageIndex() - 1, 'start');
+    }
+
+    newMessagesToastText = (count, since) => {
+        let msgSince = '';
+        if (typeof since !== 'undefined') {
+            msgSince = (
+                <FormattedMessage
+                    id='postlist.toast.since'
+                    defaultMessage=' since {date} at {time}'
+                    values={{
+                        date: (
+                            <FormattedDate
+                                value={since}
+                                weekday='short'
+                                day='2-digit'
+                                month='short'
+                            />
+                        ),
+                        time: (
+                            <FormattedTime
+                                value={since}
+                                hour='2-digit'
+                                minute='2-digit'
+                            />
+                        ),
+                    }}
+                />
+            );
+        }
+        return (
+            <React.Fragment>
+                <FormattedMessage
+                    id='postlist.toast.newMessages'
+                    defaultMessage={'{count, number} new {count, plural, one {message} other {messages}}'}
+                    values={{count}}
+                />
+                {msgSince}
+            </React.Fragment>
+        );
     }
+    renderToasts = () => {
+        const ensureShowBottom = (scrollAction, timeAtBottom) => {
+            return () => {
+                this.setState({toastShouldIgnoreBeingAtBottom: true});
+                scrollAction();
+                setTimeout(() => {
+                    this.setState({toastShouldIgnoreBeingAtBottom: false});
+                }, timeAtBottom);
+            };
+        };
+
+        const goToUnread = () => {
+            ensureShowBottom(this.scrollToUnread, toastFadeOutTimeUnread);
+        };
 
+        const jumpToLatest = () => {
+            ensureShowBottom(this.scrollToLatestMessages, toastFadeOutTimeNew);
+        };
```

Comment:
These should be turned into methods. By defining them as arrow functions in the render method, any time this component is re-rendered, it'll pass new functions as props into `UnreadToast` which will cause it to re-render when it doesn't need to